### PR TITLE
Doc fixups

### PR DESCRIPTION
**Pull request by TomAugspurger**
Found a few typos as I was going through the docs again.

I'm assuming the `array-ghost` one is a typo, didn't actually check on it.

Also added an example for doing a join of two bags of `(key, value)` pairs. Could maybe make a new function for joining two bags, but I'm not sure how general we could make it. This one is pretty specific to bags of `(key, value)` tuples.

### CODE REVIEW

**Comment by mrocklin** [comment]
We don't support joins in general, so I'm inclined not to advertise them.  Foldby brings data down to a single partition, which might be cumbersome in the general case.

**Comment by TomAugspurger** [comment]
Mmm, I didn't really appreciate that foldby outputs a single partition, but
of course it does. I think the example is still worth keeping, if only to
show another example of constructing a `binop`.

**Comment by mrocklin** [comment]
Fair enough, but we should anticipate the naive user who tries this and becomes sad when their workload blows up and they don't understand why.  There should probably be clear warnings that they should never do this in practice.

**Comment by TomAugspurger** [comment]
I updated the new `foldby` example to be an actual reduction. This should
hopefully be more appropriate.

File: dask/bag/core.py [hunk 1]
```diff
@@ -958,6 +958,37 @@ def foldby(self, key, binop, initial=no_default, combine=None,
...
+        by the ``key``. (You might be better off converting the ``Bag`` to
+        a ``dask.dataframe`` and using its groupby).
+
+        >>> import random
+        >>> import dask.bag as db
+
+        >>> tokens = list('abcdefg')
+        >>> values = range(10000)
+        >>> a = [(random.choice(tokens), random.choice(values))
+        >>>       for _ in range(100)]
```
Comment (jakirkham) [inline dask/bag/core.py]: `>>>` should be `...`. This is what is causing the CI failure.

**Comment by jrbourbeau** [comment]
@TomAugspurger I pushed a couple of commits to bring this PR back to life, hope you don't mind.

Also, the failing Travis build is unrelated to the changes in this PR.

**Comment by mrocklin** [comment]
The build failure is because we're now using Python 3 only syntax in dask.distributed.  This will likely affect all dask builds.  We should probably start skipping distributed tests in Dask in Python 2 (or drop Python 2 support for Dask)

**Comment by mrocklin** [comment]
Ah, I see that you handled this in https://github.com/dask/dask/pull/4836

**Comment by TomAugspurger** [comment]
Thanks @jrbourbeau.

**Pull request merged**